src/ape/managers/compilers.py

```diff
@@ -157,11 +157,18 @@ def compile(
                 errors.append(err)
                 continue
 
-        if errors:
+        if len(errors) == 1:
+            # If only 1 error, just raise that.
+            raise errors[0]
+
+        elif len(errors) > 1:
+            # Raise a combined error.
             formatted_errors = [f"{e}" for e in errors]
             error_message = "\n\n".join(formatted_errors)
             raise CompilerError(error_message)
```

Comment:
Wonder if python 3.11+ exception groups make sense to use when it becomes available to us

Reply:
Wow this is cool!

Idea: could try to work this in for the Python versions that support it and have it the dumb way now